jujugui/static/gui/src/app/components/deployment/deployment.js

```diff
@@ -40,15 +42,19 @@ YUI.add('deployment-component', function() {
 
       @method generateState
       @param {Object} nextProps The props which were sent to the component.
+      @param {Object} state The existing create state.
       @return {Object} A generated state object which can be passed to setState.
     */
-    generateState: function(nextProps) {
-      var state = {
-        activeComponent: nextProps.activeComponent || 'deployment-bar'
-      };
+    generateState: function(nextProps, state) {
+      if (!state) {
+        state = {};
+      }
+      state.activeComponent = nextProps.activeComponent || 'deployment-bar';
       var hasCommits = this.state ? this.state.hasCommits : false;
       var currentChangeSet = nextProps.currentChangeSet ||
           this.props.currentChangeSet;
+      var autoPlace = state.autoPlace !== undefined ?
```

Comment:
ahh could you add a comment to that effect or be explicit about checking for `false`?